.travis.yml

```diff
@@ -111,5 +112,18 @@ deploy:
   bucket: ${DEPLOY_S3_BUCKET}
   skip_cleanup: true
   local-dir: /tmp/artifacts
-  upload-dir: travis_build_artifacts/${TRAVIS_REPO_SLUG}/${TRAVIS_BRANCH}/${TRAVIS_BUILD_NUMBER}
+  upload-dir: ${ARTIFACTS_UPLOAD_PATH}
   acl: public_read
+
+after_script:
+- |
+  if [[ -v DEPLOY_S3_ACCESS_KEY ]]; then
```

Comment:
For sure we need to upload test results for both failed and successful builds (durations, stabiity). I wouldn't like to make exceptions for log files alone then.